apps/src/music/progress/MusicValidator.ts

```diff
@@ -47,62 +49,97 @@ export default class MusicValidator extends Validator {
     // Get number of sounds that have been started.
     let playedNumberSounds = 0;
 
+    // Get number of patterns that have been started.
+    let playedNumberPatterns = 0;
+
+    // Get number of chords that have been started.
+    let playedNumberChords = 0;
+
     const currentPlayheadPosition = this.player.getCurrentPlayheadPosition();
     this.getPlaybackEvents().forEach((eventData: PlaybackEvent) => {
       const length = eventData.length;
-      if (
-        eventData.when <= currentPlayheadPosition &&
-        eventData.when + length > currentPlayheadPosition
-      ) {
-        currentNumberSounds++;
 
-        if (eventData.triggered) {
+      if (eventData.type === 'sound') {
+        if (
+          eventData.when <= currentPlayheadPosition &&
+          eventData.when + length > currentPlayheadPosition
+        ) {
+          currentNumberSounds++;
+
+          if (eventData.triggered) {
+            this.conditionsChecker.addSatisfiedCondition({
+              name: MusicConditions.PLAYED_SOUND_TRIGGERED.name,
+            });
+          }
+
           this.conditionsChecker.addSatisfiedCondition({
-            name: MusicConditions.PLAYED_SOUND_TRIGGERED.name,
+            name: MusicConditions.PLAYED_SOUND_ID.name,
+            value: eventData.id,
           });
         }
 
-        this.conditionsChecker.addSatisfiedCondition({
-          name: MusicConditions.PLAYED_SOUND_ID.name,
-          value: eventData.id,
-        });
-      }
-
-      if (eventData.when <= currentPlayheadPosition) {
-        playedNumberSounds++;
+        if (eventData.when <= currentPlayheadPosition) {
```

Comment:
Could we just check `eventData.when <= currentPlayheadPosition` once outside of these nested ifs, and early return if false?

Reply:
Good idea; done.